logstash-core/src/test/java/org/logstash/config/ir/ConfigCompilerTest.java

```diff
@@ -14,4 +16,30 @@ public void testConfigToPipelineIR() throws Exception {
         assertThat(pipelineIR.getOutputPluginVertices().size(), is(1));
         assertThat(pipelineIR.getFilterPluginVertices().size(), is(0));
     }
+
+    @Test
+    public void testConfigDuplicateBlocksToPipelineIR() throws Exception {
```

Comment:
Can you document why this fix makes things deterministic?

Reply:
@andrewvc sure, let's just make this one check that each iteration produces the same result?